### PR TITLE
Feat/vendor dashboard

### PR DESCRIPTION
**Pull request by MargaritaSukhova**
## Describe your changes

- Add vendor route with children 
- Add VendorLayout component 
- Add Vendor sidebar and header components
- Add Vendor Dashboard page
- Add Vendor Card, BarChart, PieChart, PieChartLegend, OrderTable components 
- Add getDateNDaysAgo and isALeapYear utils to work with Dates
- Add useSalesHook to extract sales data for Dashboard charts
- Add tests for utils

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://2024internship.atlassian.net/browse/SCRUM-70)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [x] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.
Снимок экрана 2024-07-02 в 20 33 07
Снимок экрана 2024-07-02 в 20 33 57
Снимок экрана 2024-07-02 в 20 34 05

### CODE REVIEW

File: src/common/utils/dividePhoneNumber.ts [hunk 1]
```diff
@@ -4,9 +4,7 @@ const codeCanada: string = '+1';
 const codeUkraine: string = '+38';
 const codeCountry: string = '+62';
 
-export default function extractPhoneNumberParts(
-  phoneNumber: string | undefined
-): {
+export default function dividePhoneNumber(phoneNumber: string | undefined): {
```
Comment (lyudad) [inline src/common/utils/dividePhoneNumber.ts]: phoneNumber?: string should work

File: src/common/utils/getDateNDaysAgo.test.ts
Comment (lyudad) [inline src/common/utils/getDateNDaysAgo.test.ts:1]: move tests to separate folder under utils

Comment (MargaritaSukhova) [reply src/common/utils/getDateNDaysAgo.test.ts:1]: fixed

File: src/common/utils/dividePhoneNumber.ts (hunk 1 above)
Comment (MargaritaSukhova) [reply src/common/utils/dividePhoneNumber.ts]: fixed

**Pull request merged**